### ISSUE TITLE
Having trouble with all attributes besides Class and ID

### ISSUE THREAD

**Issue by krivaten** [Swizz/snabbdom-pragma#22]
I am having trouble getting various attributes to get rendered in the HTML. For example:

```html
// JSX
<span ariaLabel={ label }>...</span>

// and

<label for={ inputId }>...</label>

```

Compiles down to:

```js
// Compiled Ouput
Snabbdom.createElement("span", { ariaLabel: "Close" }, span)

// and

Snabbdom.createElement("label", { for: "input-id" }, label)
```

Which renders as:

```html
// Rendered HTML
<span>...</span>

// and

<label>...</label>
```

As you can see that output doesn't produce the `aria-label` and `for` attributes I defined (I've noticed the same for all other `aria` attributes as well). It looks like those attributes need to be nested under the `attrs` object in the `createElement()` execution, which I can only produce by not using JSX syntax, and instead doing something like:

```js
span({ attrs: { "aria-label": "Close" } }, [...])

// and

label({ attrs: { for: "Close" } }, [...])
```

Which compiles down to:

```js
// Compiled Output
Snabbdom.createElement("span", { attrs: { "aria-label": "Close" } }, span)

// and

Snabbdom.createElement("label", { attrs: { for: "input-id" } }, label)
```

Which renders as:

```html
// Rendered HTML
<span aria-label="Close">...</span>

// and

<label for="input-id">...</label>
```

After digging through all that, I tried the following:

```js
// JSX
function baseInput(labelString) {
    const attrs = { for: 'input' };
    return (
        <div>
            <label attrs={attrs}>{labelString}</label>
            <input id="input" />
        </div>
    );
}
```

This produces the markup I need, but for some reason that doesn't feel right.

Is there something I am doing wrong here, or is that last example the designed approach to getting the markup that I need rendered?

**Comment by jvanbruegge** [comment]
For the `aria-xxx` i just openened #23 
For the `for` you can do it like this `<label attrs-for="input" />`

**Comment by Swizz** [comment]
Hi, 
I am grateful for your interest on snabbdom-pragma.

I am sorry for the delay, I am running on low connection these past days.

You are here to talk about two problems, in fact.


**1.** Snabbdom use modules to allow the use of `class`, `props`, `attrs`, `style`, etc..
So I decided to follow this convention in snabbdom-pragma. See [Snabbdom Features](https://github.com/Swizz/snabbdom-pragma#snabbdom-features) part of the snabbdom-pragma doc.

Regarding `aria`, I chosen to consider `aria` and `data` as fake module that fallback to `props` (will be `attrs`with #23).
That make you able to use `aria` that way. See #7 
```jsx
/* written */
const vnode = <div aria-hidden={true}/>

/* Once Transpiled */
const vnode = Snabbdom.createElement('div', {'aria-hidden': true })

/* Once Executed */
const vnode = h('div', { attrs: { 'aria-hidden': true } }, [])
```

**2.** For the `for` the problem is more basic, and I am only referring to #24 to emit my POV.
But in a dreamed world, all attributes that is not gonna be a module is added to the `props` module.
But some of them need to be consumed with the `attrs` module instead of the `props` one.

You could still use the `attrs` module to force this, but I will consider merging #24 to avoid this.

```jsx
/* written */
const vnode = <label attrs-for={ inputId }>...</label>

/* Once Transpiled */
const vnode = Snabbdom.createElement('input', {'attrs-for': inputId }, '...')

/* Once Executed */
const vnode = h('div', { attrs: { 'for': inputId } }, ['...'])
```

**Comment by Swizz** [comment]
All this point is now fixed thanks to #23 and #24 and Npm published with [v2.6.0](https://github.com/Swizz/snabbdom-pragma/releases/tag/2.6.0).

If your issue is not fixed feel free to reopen it.
